app/src/main/java/org/greenstand/android/TreeTracker/viewmodels/NewTreeViewModel.kt

```diff
@@ -52,7 +51,21 @@ class NewTreeViewModel(
 
     suspend fun createTree(note: String) {
 
-        val newTree = createNewTree(note, photoPath!!, newTreeUuid!!)
+        val newTree = Tree(
+            treeUuid = newTreeUuid!!,
+            planterCheckInId = user.planterCheckinId ?: -1,
+            content = note,
+            photoPath = photoPath!!
+        )
+
+        val absoluteStepCount = user.absoluteStepCount ?: 0
+        val lastStepCountWhenCreatingTree = user.absoluteStepCountOnTreeCapture ?: 0
+        // Delta step count is the difference between the absolute count at the time of capturing
+        // a tree minus the last absolute step count recorded when capturing a previous tree. This
+        // is the indicator for the number of steps taken between two trees.
+        val deltaSteps = absoluteStepCount - lastStepCountWhenCreatingTree
```

Comment:
Looking at the docs it seems the step counter shows total steps since we started tracking since the device has last turned on. That sounds like if we plant a tree, then wait a week we can have a delta of many miles for our next tree. Is this correct?

Reply:
That is true. This came up in our dicussion in the team meeting and @deepwinter mentioned it is fine for the very first tree captured in a session. They can factor that in.

Reply:
Yes this is fine.  We will catch that in the pipeline.